drivers/mpu9250/MPU9250.cpp

```diff
@@ -522,15 +433,18 @@ void MPU9250::_measure()
 
 				// Initialize the temperature logic.
 				_last_temp_c = temp_c;
+				DF_LOG_INFO("IMU temperature initialized to: %f", temp_c);
 				_temp_initialized = true;
 			}
-
 		} else {
 			// Once initialized, check for a temperature change of more than 2 degrees which
 			// points to a FIFO corruption.
 			if (fabsf(temp_c - _last_temp_c) > 2.0f) {
-				DF_LOG_ERR("FIFO corrupt");
+				DF_LOG_ERR(
+						"FIFO corrupt, temp difference: %f, last temp: %f, current temp: %f",
+						fabs(temp_c - _last_temp_c), (double)_last_temp_c, (double)temp_c);
 				reset_fifo();
+				_temp_initialized = false;
```

Comment:
Not sure what you mean by "have this init flag."  The value of the variable is initialized in the object ctor.